### ISSUE TITLE
Getting a script error while running game from editor doesn't show error message.

### ISSUE THREAD

**Issue by Khatoblepas** [adventuregamestudio/ags#1034]
**Describe the bug**
When testing the game in the editor, if you get a script error, instead of popping up an error message about a script error, it just closes the Output window and changes focus to the Call Stack instead. Using 256 colour mode with the software renderer in windowed mode.

**AGS Version**
3.5.0.23

**Expected behavior**
As with 3.2.1, if the game aborts with an error, a pop up window should appear with the script error bug, and change to the script editor to the highlighted line.

**Desktop (please complete the following information):**

    OS: Windows
    Version: 7

**Comment by ivan-mogilko** [comment]
I do not observe this error. Does this happen in case of any error? What if you try non-8-bit game?

**Comment by Khatoblepas** [comment]
I tried making a new project that was 32bit and it does the same thing. Testing by using AbortGame to force an error pop up - the game window doesn't close either.

**Comment by ivan-mogilko** [comment]
Do you close the game by pressing "Stop" button on toolbar?

Is it correct, that it also does not open the script pane, nor highlights the script line?

Could you check, when this happens, in the File menu - are Open, Save and Exit commands disabled or stay active?

**Comment by Khatoblepas** [comment]
They are disabled. I've found the source of the issue - the little error box appears to the left of the AGS window, When maximised on my main monitor, the error box opens off screen and can't be clicked. When I make the AGS window smaller or move it to the monitor to the right, the error box appears. Maybe the window is spawned in the wrong place relative to the main window?

**Comment by ivan-mogilko** [comment]
Alright, so this is a multiple monitors issue.

**Comment by ivan-mogilko** [comment]
![Screenshot from Khatoblepas](https://user-images.githubusercontent.com/1833754/73611297-99eb7000-45f1-11ea-8a10-307310b7ba23.png)

**Comment by ivan-mogilko** [comment]
Showing error message is handled by this code in Scintilla: https://github.com/adventuregamestudio/ags/blob/dc9f95ad7aa36bf17dbdcad2903f1b81335367ce/Editor/AGS.Editor/Panes/ScintillaWrapper.cs#L408

perhaps, it does not account for multiple monitors, or has some oversights.

**Comment by ericoporto** [comment]
A few lines below there's this:

`Point mainWindowOffset = this.PointToScreen(new Point(0, 0));`

Assuming Windows Forms works and PointToScreen comes from there, perhaps ignoring the offset from scintilla and using the above is enough to get a position relative to AGS Window on screen if this is the intent. I don't have multiple monitors to test and verify. :/

**Comment by morganwillcock** [comment]
I think I can recreate it. It looks like the smaller error window tries to locate itself based on an offset from the line that is highlighted with the error, and if you have moved the window (probably to see the game on a smaller monitor, or to a different monitor that is smaller) this location may be off-screen. Dragging the full Editor window back into view doesn't move the smaller window, so it just gets let off-screen where it cannot be clicked, and choosing to 'cascade all windows' (in the Taskbar) doesn't bring it back either.

It is more likely to be a problem where the script line is near the end of the script file, because this means that the line is unable to be centered (because there is no more space remaining in the script buffer that would allow further scrolling).

To fix, I'm not 100% sure what the purpose of this window actually is. Can we just remove it? The call stack is forcibly shown anyway, so I'm not sure why the error or abort message isn't just shown there and you stop the game with the regular stop button. Or maybe the script buffer supports some kind of overlay message, similar to the autocomplete pop-up.

**Comment by ivan-mogilko** [comment]
> To fix, I'm not 100% sure what the purpose of this window actually is.

Perhaps it was to make it more clear to the user what happened and how to proceed after getting an error.

**Comment by persn** [comment]
I'll vote yes to less pop up dialogs any day as long as we can make sure the alternative approach gives the user feedback in a satisfactory manner

**Comment by ivan-mogilko** [comment]
Then this is also a question of whether to fix this as a part of patch, or change the way error is reported in new version, or both (separately).

**Comment by morganwillcock** [comment]
There are separate state issues with this window, that if you close the script editing buffer it persists, you can generate another one by revisting the error (i.e. double click the line in the call stack listing), and now you will have two. If you the new one to close the game and then click the old one you get stuck in a state where the game isn't running but the script editor is the run/debug controls are only showing pause and stop, and all the run options are disabled.

I've had a quick try to show the error message as a 'tip', but I don't think the script editor is in the correct state to show it (or perhaps it is cleared instantly by the autocomplete processing) as it didn't seem to appear.

**Comment by ivan-mogilko** [comment]
@morganwillcock 
> I think I can recreate it. It looks like the smaller error window tries to locate itself based on an offset from the line that is highlighted with the error, and if you have moved the window (probably to see the game on a smaller monitor, or to a different monitor that is smaller) this location may be off-screen.

Could you clarify, how do you reproduce this problem?

**Comment by ivan-mogilko** [comment]
Ok, I managed to reproduce this by moving editor window mostly offscreen and then triggering script error.

Also, double-clicking the item in callstack tries to move error window to it. But if you unpin callstack pane, then it cannot do that properly....

**Comment by morganwillcock** [comment]
It doesn't seem too bad to just show the error in the output panel. The tab takes precedence over the call stack tab, I guess because the content is added later. Maybe if the output panel becomes multiline aware with better handling of the icon (and maybe passed the filename and line number to use existing columns) this would be OK. Build errors/status go to this panel as well, so it is already not exclusively for script compilation errors.

![panel](https://user-images.githubusercontent.com/14958747/75077807-b77a6e00-54fb-11ea-89e5-14b1fcfc501c.png)

**Comment by ivan-mogilko** [comment]
> It doesn't seem too bad to just show the error in the output panel. 

I agree, that's the most logical place to print them out of all existing panes.

**Comment by morganwillcock** [comment]
I have a rough version that works, including setting the file and line columns, but unfortunately there is no text wrap on this control so longer error messages will be problematic. So the 'correct' solution is probably to re-implement the entire output panel for longer messages and different types of messages. That is a lot of work though, and my time is limited at the moment.

**Comment by persn** [comment]
Maybe the output panel should be just a plain textfield with user typing disabled, similar to Visual Studio

**Comment by morganwillcock** [comment]
It does let you jump to the line in the buffer by double clicking it, which I guess is pretty useful. But I'm not sure how the information ends up in this panel is according to the original design either. The class that is used to pass the data around is actually an exception class, but in a lot of cases it is just used to pass the data and not thrown (and I don't think the design was ever to throw it high enough to reach the panel). It probably just happened to have the required fields and so got re-used, but was originally only meant to be used during script compilation. So ideally there needs to be a new control for message output, plus some kind of intermediate formatter that can be used from where thrown script compiler exceptions are caught, as well as for generic messages (which may or may not have a script reference).

**Comment by ericoporto** [comment]
@morganwillcock this new control just to receive runtime error output, is there anyone in particular from an IDE you think it's relevant? The most similar I could think was logcat from Android Studio but it appears it's still very different from what you suggested.

**Comment by morganwillcock** [comment]
It would have to be multipurpose to take over from the existing one and incorporate the runtime error output. Just trying to fix the position of the existing error window didn't look easy and it seems to potentially break the editor state if you happen to click things in the wrong order, so I'm not sure there is an easier fix for everything that doesn't completely remove the error window. Maybe just showing the window in the Task Bar is a temporary improvement, so at least if the window isn't visible someone has a clue of what is going on - but that doesn't solve the issue really.

**Comment by ericoporto** [comment]
the output design is here: [Editor/AGS.Editor/GUI/OutputPanel.Designer.cs](https://github.com/adventuregamestudio/ags/blob/master/Editor/AGS.Editor/GUI/OutputPanel.Designer.cs)

I noticed it uses a ListView, which doesn't support multiline. If it is replaced by a DataGridView, than it can have multiline support. Alternatively, no multiline but on mouse over the tooltip could show the content rendered with multiline.
